haystack/components/builders/chat_prompt_builder.py

```diff
@@ -211,12 +213,16 @@ def _validate_variables(self, provided_variables: Set[str]):
         :raises ValueError:
             If no template is provided or if all the required template variables are not provided.
         """
-        missing_variables = [var for var in self.required_variables if var not in provided_variables]
+        if self.required_variables == "*":
+            required_variables = sorted(self.variables)
```

Comment:
Any specific reason why we want to sort the variables here? self.required_variables is also not sorted right?

Reply:
Yeah specifically here b/c `self.variables` is a list but is created from a Set. So in this specific scenario it's possible for the order of required_variables to change when inheriting from `self.variables` which causes the Error message to be inconsistent. We look for an exact string match for one of the tests so without the sort the test would become unreliable.